src/system/api/rqidApi.ts

```diff
@@ -114,12 +116,50 @@ export class Rqid {
       result.splice(0, 0, ...worldDocuments);
     }
 
+    const distinctWorldRqids: string[] = result
+      .map((d) => d.data?.flags?.rqg?.documentRqidFlags?.id)
+      .filter(function (value, index, self) {
+        return self.indexOf(value) === index;
+      });
+
+    // find the best results in the world
+    if (best) {
+      console.log("Distinct World Rqids: ", distinctWorldRqids);
+      result = [];
+      for (const rqid of distinctWorldRqids) {
+        const best = await Rqid.fromRqid(rqid);
+        if (best) {
+          result.push(best);
+        }
+      }
+    }
+
     if (["match", "all", "compendiums"].includes(scope)) {
-      const compendiaDocuments = await Rqid.documentsFromCompendia(
+      let compendiaDocuments = await Rqid.documentsFromCompendia(
         rqidRegex,
         rqidDocumentType,
         lang
       );
+
+      if (best) {
+        // list of all the rqids from the compendia that were not already found in the world.
+        const distinctCompendiaRqids = compendiaDocuments
+          .map((d) => d.data?.flags?.rqg?.documentRqidFlags?.id)
+          .filter(function (value, index, self) {
+            return self.indexOf(value) === index;
+          })
+          .filter(
+            (rqid) => !distinctWorldRqids.includes(rqid)
+          );
+        compendiaDocuments = [];
+        for (const rqid of distinctCompendiaRqids) {
+          const best = await Rqid.fromRqid(rqid);
```

Comment:
Maybe change the variable name to something else than best to avoid confusion with the outer best variable.

Reply:
Done

Reply:
The parameter is now "findOnlyBest" and the loop variables are now "bestWorldDocFound" and "bestCompendiaDocFound".